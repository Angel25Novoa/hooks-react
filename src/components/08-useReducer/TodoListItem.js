import React from 'react'
import PropTypes from 'prop-types';
import './styles.css'

const TodoListItem = ({todo, index, handleDelete, handleToggle}) => {
  return (
    <li className='element'>
      <p
        onClick={() => handleToggle(todo.id)}
        className={todo.done ? 'done' : 'notDone'}
      >
        {index + 1 + ':'}{todo.name}
      </p>
      <button
        className='btn btn-danger'
        onClick={ () => handleDelete(todo.id) }
      >
        Delete
      </button>
    </li>
  )
}

TodoListItem.propTypes = {
  todo: PropTypes.object.isRequired,
  index: PropTypes.number.isRequired,
  handleDelete: PropTypes.func,
  handleToggle: PropTypes.func
}

export default TodoListItem