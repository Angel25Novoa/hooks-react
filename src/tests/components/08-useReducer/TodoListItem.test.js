import React from 'react'
import {shallow} from 'enzyme'  
import TodoListItem from '../../../components/08-useReducer/TodoListItem'
import {demoState} from '../../fixtures/demoTodos'

//usar demoTodos para testear

describe((`TodoListItem component tests`), () => {

  const handleDelete = jest.fn()
  const handleToggle = jest.fn()
  const todo = demoState[0]
  const wrapper = shallow(
    <TodoListItem 
      todo={todo}
      index={0}
      handleDelete = {handleDelete}
      handleToggle = {handleToggle}
    />)
  test(`TodoListItem component should render without errors`, () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('TodoListItem can call handleDelete', () => {
    wrapper.find('button').simulate('click')
    expect(handleDelete).toHaveBeenCalledWith(todo.id)
    //jest.fn()
    //const handleDelete = jest.fn()
    //toHaveBeenCalledWith
  })
  test('TodoListItem can call handleToggle', () => {
    wrapper.find('p').simulate('click')
    expect(handleToggle).toHaveBeenCalledWith(todo.id)
    //jest.fn()
    //const handleDelete = jest.fn()
    //toHaveBeenCalledWith
  })
  test('Should render the todo', () => {
    expect(wrapper.find('p').text().trim()).toBe(`1:${todo.name}`)
  })
  test('Should toggle the todo', () => {
    const todoTest = {...todo, done: true}
    const wrapper = shallow(
      <TodoListItem 
        todo={todoTest}
        index={0}
        handleDelete = {handleDelete}
        handleToggle = {handleToggle}
      />)
    wrapper.find('p').simulate('click')
    expect(wrapper.find('p').hasClass('done')).toBe(true)
  })
})