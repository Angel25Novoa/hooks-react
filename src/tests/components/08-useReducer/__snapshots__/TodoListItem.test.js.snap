// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListItem component tests TodoListItem component should render without errors 1`] = `
<li
  className="element"
>
  <p
    className="notDone"
    onClick={[Function]}
  >
    1:
    Hacer la compra
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
